server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -207,6 +211,19 @@ public DocTableInfo(RelationName ident,
             .toList();
     }
 
+
+    /**
+     * <p>Contains all information to parse the source:</p>
+     * @param referencesByOid is needed to rewrite OID-s to column names.
+     * OID-s is casted to String since source keys are stored as String
+     * and it's cheaper to cast long->String once and do OID rewrites without per-row casting.
+     * @param droppedColumns is needed to hide dropped columns from the source.
+     * @param versionCreated is needed to ignore DROP COLUMN related logic for old tables.
+     */
+    public static record SourceParsingMetadata(Map<String, Reference> referencesByOid,
+                                               Set<String> droppedColumns,
+                                               Version versionCreated) {}
+
```

Comment:
`Version` - to fall back to the old logic on 5.4 tables (if an old was restored to the 5.5 node) 

`Set<String> droppedColumns` - already used in SourceParser but not for _raw lookup, but will be used for hiding dropped columns in _raw in https://github.com/crate/crate/pull/14645 (this PR is the first commit).

`Map<String, Reference> referencesByOid` - Will be used for rewriting OID-s on raw lookup, also `values` is `references` and they are needed to register ALL columns on raw lookup (https://github.com/crate/crate/pull/14645/commits/fe5a43f8d482c666c7915bf0a051e3e51525b925) so double use of it. 

Also, oid key is string to avoid per-row cast of String_from_source to Long_in_map_created_once